orttraining/orttraining/core/optimizer/megatron_transformer.cc

```diff
@@ -409,14 +534,19 @@ Status MegatronTransformer::TransformBARTMLP(Graph& graph, bool& modified,
       biasgelu_node.GetOutputEdgesCount() != 1) {
     return skip_status;
   }
-  Node& dropout_node = *graph.GetNode(biasgelu_node.OutputNodesBegin()->Index());
-  if (!IsExpectedOpAndProvider(dropout_node, dropout_info, provider_type)) {
-    return skip_status;
+
+  // Either Dropout->Matmul or just Matmul
```

Comment:
the onnx graph added to the PR for testing has this pattern. It could be from a different commit/hyperparameters than what the transforms were written for

Reply:
should it be the same with the one I am running now? I guess you exported it in inference model, then we dropped Dropout node?

Reply:
The onnx exported graph has a Dropout node but with a ratio=0.0, which causes the ORT transforms to drop this node as it is equivalent to Identity. Thus the pattern that megatron transform sees doesnt have this Dropout. It is exported via save_as_onnx which should export in training mode, and there are other Dropout nodes present in the graph that are valid. I looked at the torch code and looks like there is a parameter(attention_dropout) that defines the Dropout ratio with a default value of 0, which might be the difference between the huggingface commits that are being used. In any case, since the model added here is just for testing the checkpointing functionality, it should be okay.